### PR TITLE
Add PSP for the controller Deployment

### PR DESCRIPTION
**Pull request by george-angel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
Adds PSP for the controller Deployment. Without it, clusters with default PSP having basic permissions, the controller will not be allowed to run.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
No

```release-note
NONE
```

### CODE REVIEW

File: deploy/kubernetes/base/controller/psp.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  seLinux:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  runAsUser:
+    rule: RunAsAny
+  fsGroup:
+    rule: RunAsAny
+  volumes:
+    - "configMap"
```
Comment (mattcary) [inline deploy/kubernetes/base/controller/psp.yaml:15]: Why allow configMaps if the controller isn't currently using any?

**Review by mattcary — COMMENTED** [review]
Thanks! This is a good idea, thanks for the PR.

One small comment above.

File: deploy/kubernetes/base/controller/psp.yaml (hunk 1 above)
Comment (george-angel) [reply deploy/kubernetes/base/controller/psp.yaml]: @mattcary good catch, no reason. I just copied our patch, removed.

**Comment by mattcary** [comment]
/lgtm
/approve

Thanks again for your contribution!

**Comment by mattcary** [comment]
I think you need to update the description as described [here](https://github.com/kubernetes/community/blob/master/contributors/guide/release-notes.md#applying-a-release-note) in order to get the release note nag label removed.

**Pull request merged**